apps/userdashboard/api.py

```diff
@@ -84,6 +84,16 @@ def update(self, request, *args, **kwargs):
 
         return super().update(request, *args, **kwargs)
 
+    @action(detail=True, methods=["PATCH"])
+    def set_show_ai_report(self, request, **kwargs):
+        if "show_ai_report" in request.data:
+            comment = self.get_object()
+            comment.ai_report.show_ai_report = request.data["show_ai_report"]
+            comment.ai_report.save()
```

Comment:
i'd add a check here if the incoming data is type of bool, since we circumvent the validation with the comment-moderator-feedback serializer.